cl/phase1/network/services/batch_signature_verification.go

```diff
@@ -54,27 +53,26 @@ func (b *BatchSignatureVerifier) AddVerification(aggregateVerificationData *Aggr
 // and verify them together - running all the final functions afterwards
 func (b *BatchSignatureVerifier) Start() {
 	ticker := time.NewTicker(batchCheckInterval)
+	defer ticker.Stop()
 	aggregateVerificationData := make([]*AggregateVerificationData, 0, 128)
 	for {
 		select {
 		case <-b.ctx.Done():
 			return
 		case verification := <-b.verifyAndExecute:
-			b.size += uint64(len(verification.Signatures))
 			aggregateVerificationData = append(aggregateVerificationData, verification)
-			if b.size > BatchSignatureVerificationThreshold {
+			if len(aggregateVerificationData) > BatchSignatureVerificationThreshold {
```

Comment:
I wanted to use `b.size` because we don't know how many signatures we are verifying otherwise because in each `AggregateVerificationData` there might be 1 transaction or 3 (or in the future 100 🤷 ).

That's why I was looking into it with `len(verification.Signatures)` and was verifying when I had specific number of transactions and not `AggregateVerificationData`.

But this should also work

Reply:
I think this tells us better about our memory cap. as you say, if you have b.size = 100 but have actually 1000 signatures, then it might not be expected